### PR TITLE
Logging shutdown hack

### PR DESCRIPTION
**Pull request by jasontedor**
Log4j has a bug where on shutdown it ignores that its use of JMX might
be disabled. This leads to Log4j attempting to access JMX leading to a
security exception. This pull request intentionally introduces jar hell
with the Server class to work around this bug until a fix is a
released. We also move Log4j shutdown from Node#stop to Node#close as
the former was too early (but was masked by the above Log4j issue) given
that we also try to log in Node#close.

Relates #20304

### CODE REVIEW

**Comment by jasontedor** [comment]
And finally there are clean shutdowns again:

```java
^C[2016-09-08T12:55:47,698][INFO ][node                     ] [kpxKW1m] stopping ...
[2016-09-08T12:55:47,711][INFO ][node                     ] [kpxKW1m] stopped
[2016-09-08T12:55:47,711][INFO ][node                     ] [kpxKW1m] closing ...
[2016-09-08T12:55:47,719][INFO ][node                     ] [kpxKW1m] closed
$
```

**Comment by jasontedor** [comment]
retest this please

**Comment by jasontedor** [comment]
retest this please

File: core/src/main/java/org/apache/logging/log4j/core/jmx/Server.java [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+import org.apache.logging.log4j.status.StatusLogger;
+import org.apache.logging.log4j.util.PropertiesUtil;
+
+/**
+ * Creates MBeans to instrument various classes in the log4j class hierarchy.
+ * <p>
+ * All instrumentation for Log4j 2 classes can be disabled by setting system property {@code -Dlog4j2.disable.jmx=true}.
+ * </p>
+ */
+public final class Server {
```
Comment (jaymode) [inline core/src/main/java/org/apache/logging/log4j/core/jmx/Server.java]: maybe we can also link to the issue / version this affects here?

**Comment by jaymode** [comment]
LGTM

**Comment by abeyad** [comment]
LGTM2

**Comment by abeyad** [comment]
@jasontedor the test LGTM

**Comment by jaymode** [comment]
test LGTM2

File: core/src/main/java/org/apache/logging/log4j/core/jmx/Server.java (hunk 1 above)
Comment (jasontedor) [inline core/src/main/java/org/apache/logging/log4j/core/jmx/Server.java]: There's a [link](https://github.com/elastic/elasticsearch/pull/20389/files#diff-cf110840ff1265b6b449a57fc3f34099R285) the jar hell exemption, is that sufficient?

Comment (jaymode) [inline core/src/main/java/org/apache/logging/log4j/core/jmx/Server.java]: yes

**Comment by jasontedor** [comment]
Thanks again @abeyad and @jaymode.

**Pull request merged**